charts/keycloak/templates/networkpolicy.yaml

```diff
@@ -0,0 +1,46 @@
+{{- if .Values.networkpolicy.enabled }}
+apiVersion: "networking.k8s.io/v1"
+kind: "NetworkPolicy"
+metadata:
+  name: "{{ template "keycloak.fullname" . }}"
+  labels:
+    {{- include "keycloak.labels" . | nindent 4 }}
+    {{- with .Values.networkpolicy.labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  policyTypes:
+    - "Ingress"
+  podSelector:
+    matchLabels:
+      {{- include "keycloak.selectorLabels" . | nindent 6 }}
+  ingress:
+    {{- if .Values.networkpolicy.from }}
+    - from:
+        {{- toYaml .Values.networkpolicy.from | nindent 8 }}
+      ports:
+        - protocol: TCP
+          port: {{ .Values.service.httpPort }}
+        - protocol: TCP
+          port: {{ .Values.service.httpsPort }}
```

Comment:
Use named ports. Also add the management port.

Reply:
Named port on network policies is not support by all k8s networking layer. 
For exemple, my current cluster doesn't support it (weave-net, old version :sad: ), if someone define a netpol with a named port this break the entire cluster networking.

Since this is a chart (aka managed by value files), name or configured port is quite the same, no ?
Or maybe I can add an option to manage this (useNamedPort: boolean). 
Let me know what is your preference ? 

For managed port, this is intentional. 
I was thinking that port should only be used by Keycloak pods. That's probabily wrong. 
Should I add an option for each port to let user define which port is allowed on this `from` user defined policies ? 

`managementPort` is already present on keycloak to keycloak pods communication policy (below)

Reply:
I didn't try it but it is documented that named ports work:

```
$ k explain networkpolicy.spec.ingress.ports
KIND:     NetworkPolicy
VERSION:  networking.k8s.io/v1

RESOURCE: ports <[]Object>

DESCRIPTION:
     List of ports which should be made accessible on the pods selected for this
     rule. Each item in this list is combined using a logical OR. If this field
     is empty or missing, this rule matches all ports (traffic not restricted by
     port). If this field is present and contains at least one item, then this
     rule allows traffic only if the traffic matches at least one port in the
     list.

     NetworkPolicyPort describes a port to allow traffic on

FIELDS:
   port	<string>
     The port on the given protocol. This can either be a numerical or named
     port on a pod. If this field is not provided, this matches all port names
     and numbers.

   protocol	<string>
     The protocol (TCP, UDP, or SCTP) which traffic must match. If not
     specified, this field defaults to TCP.
```

Reply:
Yes, totally agreed but their is some issues on implementations : https://github.com/weaveworks/weave/issues/3785. 
If I put named port in this PR a lot of users cannot be able to use this networkpolicies template (like me ;-) )